src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -433,8 +450,18 @@ public final void run() {
                         // in an immediately ready selection key, hence we use the non-blocking form
                         selected = selector.selectNow();
                     } else {
-                        selected = selector.select(); // WINDOWS!!! Waiting the whole timeout anyway if timeout specified?!?
+                        // On Windows the select(timeout) operation ALWAYS waits for the timeout,
+                        // so we workaround it by WindowsIOHubSelectorWatcher
+                        if (Launcher.isWindows()) {
+                            selectorIsWaiting = true;
```

Comment:
none of these are volatile and there is no synchronization, the other thread will not be guaranteed to see them

Reply:
Added the task to the PR to make it explicit